lite/backends/nnadapter/nnadapter/core/operation/concat.cc

```diff
@@ -22,6 +23,42 @@
 namespace nnadapter {
 namespace operation {
 
+template <typename T>
+static void concat_offline_calc(
+    const std::vector<int32_t*> inputs,
+    const std::vector<std::vector<int64_t>> inputs_dims,
+    const int axis,
+    int32_t* output,
+    std::vector<int64_t> output_dims) {
```

Comment:
```suggestion
static void ConcatOfflineCalc(
    const std::vector<T*>& inputs,
    const std::vector<std::vector<int32_t>>& inputs_dims,
    const int axis,
    T* output) {
```

- 函数名称单词首字母大写
- 模板类型T表示数据类型，那么函数中相应的类型都应该用T表示
- nnadapter中维度都用 int32_t
- const vector 一般传引用
- output_dims 涉及到的计算比较简单，而且不是函数功能实现所必须的参数

Reply:
> output_dims 涉及到的计算比较简单，而且不是函数功能实现所必须的参数

那应该还是需要传入一个out_concat_axis类似的参数吧
  const int out_concat_axis = output_dims[axis];

Reply:
> 那应该还是需要传入一个out_concat_axis类似的参数吧 const int out_concat_axis = output_dims[axis];

可以从 inputs_dims 相应 axis 维度求和得到吧

Reply:
也行，我本来是想直接复用之前infer_out_shape的结果。